casper/src/main/resources/PoS.rhox

```diff
@@ -344,17 +405,25 @@ new  PoS,
           }
         } |
 
-        contract deposit (@fromUserPk, @amount, returnCh) = {
+        contract deposit (@fromUserPk, @amount, @activeValidators, returnCh) = {
           new vaultCh,
               revAddressCh,
+              indexSender,
+              getBlockData(`rho:block:data`), blockDataCh,
               authKeyCh in {
             @RevVault!("deployerAuthKey", *authKeyCh) |
+            getBlockData!(*blockDataCh) |
             for (@authKey <- authKeyCh) {
               revAddressOps!("fromPublicKey", fromUserPk, *revAddressCh) |
               for (@fromRevAddress <- revAddressCh) {
                 @RevVault!("findOrCreate", fromRevAddress, *vaultCh) |
-                for (@(true, fromVault) <- vaultCh) {
-                  @fromVault!("transfer", posRevAddress, amount, authKey, *returnCh)
+                for (@(true, fromVault) <- vaultCh; _, _, @sender <- blockDataCh) {
```

Comment:
I would put the receive from blockDataCh closer to the getBlockData!(*blockDataCh) for better readability. One next to the other is the easier to match visually